app/src/main/java/space/taran/arknavigator/mvp/presenter/adapter/tagsselector/TagsSelectorPresenter.kt

```diff
@@ -261,51 +295,96 @@ class TagsSelectorPresenter(
     }
 
     private fun filterTags() {
-        availableTagsForDisplay = availableTags
-            .filter { it.startsWith(filter, false) }
+        availableTagsForDisplay = availableTagItems
+            .filter {
+                when (it) {
+                    is TagItem.DefaultTagItem -> {
+                        it.tag.startsWith(filter, false)
+                    }
+                    is TagItem.KindTagItem -> {
+                        kindToString[it.kind]!!.startsWith(filter, false)
+                    }
+                }
+            }
 
-        unavailableTagsForDisplay = unavailableTags
-            .filter { it.startsWith(filter, false) }
+        unavailableTagsForDisplay = unavailableTagItems
+            .filter {
+                when (it) {
```

Comment:
Code duplicate. Maybe we should implement something like `KindTagItem extends Stringlike` where `Stringlike { fun gegtString(): String { ... } }`?

Reply:
Replaced with `filterTagItem` function